update permissions help

alxnddr opened the pull request:
Epic https://github.com/metabase/metabase/issues/29491

### Description

Update permissions help sidebar content.

### How to verify

1. As an admin go to http://localhost:3000/admin/permissions
2. Check data permissions help
3. Check collection permissions help
4. Check application permissions help (EE-only)

### Demo

<img width="1362" alt="Screen Shot 2023-06-07 at 9 56 41 PM" src="https://github.com/metabase/metabase/assets/14301985/632aa34f-e237-4e5e-91cb-ce683f4dc110">

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

alxnddr committed:
update permissions help

alxnddr commented:
@vbenedetti @luizarakaki on the mockups we add `(Pro)` to Impersonated and Sandboxed permissions but not to the Block permission which is also only for EE users. Should we add it there as well?

Also the phrase "Only available in certain Metabase plans." that goes after the Block permission without the upgrade link and after Sandboxed with the upgrade link. The Impersonated permission description does not have this phrase — should it be there?

<img width="598" alt="Screen Shot 2023-06-08 at 11 08 43 PM" src="https://github.com/metabase/metabase/assets/14301985/7eb02292-2581-4d85-8615-e74d0df7b5bd">

JesseSDevaney commented on frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx:
```diff
@@ -0,0 +1,126 @@
+            <PermissionHelpDescription
+              icon="close"
+              iconColor="danger"
+              name={t`Block (Pro)`}
```
I agree that block should have a (Pro) descriptor, but it was not there in the design docs. I assume you confirmed this with product already?
```suggestion
              name={t`Block`}
```

JesseSDevaney commented on frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx:
```diff
@@ -0,0 +1,126 @@
+              description={t`Let's you specify row and column-level permissions. Can be set up via user attributes and SSO.`}
+            />
+
+            <Text>{t`Only available in certain Metabase plans.`}</Text>
```
I saw the design docs specified having an `Upgrade to Pro` link. Were you waiting to add this to see if product wanted it on both accordions under 'Only available in certain Metabase plans'?

JesseSDevaney reviewed (approved):
LGTM - I just have two questions about some differences between the design doc and implementation

npfitz commented on e2e/test/scenarios/permissions/admin-permissions.cy.spec.js:
```diff
@@ -734,3 +734,31 @@ describeEE("scenarios > admin > permissions", () => {
+
+  it("shows permissions help", () => {
+    cy.visit("/admin/permissions");
+    cy.get("main").within(() => {
```
Is there a more specific selector we can use here?

alxnddr replied:
Good catch, updated where I could

luizarakaki commented:
@alxnddr Yes for both

- Please add `(Pro)` for block.
- Add the upgrade link after Block and Impersonated

alxnddr committed:
update help copy, reexport mantine components

alxnddr commented:
@npfitz @JesseSDevaney thanks for reviews, I updated the PR and also removed `mantine` direct imports outside of `metabase/ui`.

@luizarakaki I update the copy, thanks for clarification, please have a look. I don't show the upgrade link and the copy "Only available in certain Metabase plans." if user runs Metabase EE already, is that ok?

luizarakaki commented:
Yeah, makes sense to hide for pro/ee

vbenedetti commented:
@alxnddr - I just checked the branch, the upgrade we did for the permissions help looks great. I'm very happy that it's coming live. 
I know this is not explicitly said on the spec, but playing with the branch, I wish the behavior when navigating from Data to Collections and Application Permissions affected the help differently.

Current: In Data Permissions I open the Permissions help, then I Click on collections. The permission help closes and I have to open again to notice the contextual change.

preferred: When clicking on Collections, if the help is open, it remains open and the content changes to match the new context. 

Could we make this little update?
Thanks!

JesseSDevaney reviewed (approved):
:gem: Looking good! I just have a few questions about variable naming.

luizarakaki commented:
Played with the branch today and it is looking good. I'd like to see the improvement @vbenedetti proposed. It makes sense to keep the sidebar open while we change tabs.

alxnddr committed:
keep help reference open when switching tabs

alxnddr committed:
review

vbenedetti reviewed (approved):
Looks great! thanks for adding the last minute improvement.

alxnddr committed:
update feature detection logic and the plans link
